1.segment_images/scripts/1.segment_cells_organoids.py

```
@@ -0,0 +1,262 @@
+from cellpose import models
+
+cellpose_io.logger_setup()
+import torch
+from cellpose.io import imread
+from PIL import Image
+from skimage import io
+
+use_GPU = torch.cuda.is_available()
+
+# set import path
+sys.path.append(str(pathlib.Path("../../utils/").resolve()))
+
+# check if in a jupyter notebook
+try:
+    cfg = get_ipython().config
+    in_notebook = True
+except NameError:
+    in_notebook = False
+
+print(in_notebook)
+
+
+# In[2]:
+
+
+if not in_notebook:
+    # set up arg parser
+    parser = argparse.ArgumentParser(description="Segment the nuclei of a tiff image")
+
+    parser.add_argument(
+        "--input_dir",
+        type=str,
+        help="Path to the input directory containing the tiff images",
+    )
+    parser.add_argument(
+        "--window_size", type=int, help="Size of the window to use for the segmentation"
+    )
+    parser.add_argument(
+        "--clip_limit",
+        type=float,
+        help="Clip limit for the adaptive histogram equalization",
+    )
+
+    args = parser.parse_args()
+    window_size = args.window_size
+    clip_limit = args.clip_limit
+    input_dir = pathlib.Path(args.input_dir).resolve(strict=True)
+
+else:
+    input_dir = pathlib.Path("../../data/z-stack_images/C4-2/").resolve(strict=True)
+    window_size = 3
+    clip_limit = 0.1
+
+mask_path = pathlib.Path(f"../processed_data/{input_dir.stem}").resolve()
+mask_path.mkdir(exist_ok=True, parents=True)
+
+
+# ## Set up images, paths and functions
+
+# In[3]:
+
+
+image_extensions = {".tif", ".tiff"}
+files = sorted(input_dir.glob("*"))
+files = [str(x) for x in files if x.suffix in image_extensions]
+
+
+# In[4]:
+
+
+# find the cytoplasmic channels in the image set
+for f in files:
+    if "405" in f:
+        nuclei = io.imread(f)
+    elif "488" in f:
+        cyto1 = io.imread(f)
+    elif "555" in f:
+        cyto2 = io.imread(f)
+    elif "640" in f:
+        cyto3 = io.imread(f)
+    else:
+        print(f"Unknown channel: {f}")
```

Do you think that the unknown channels should raise and error instead of printing that it is? There is the potential risk that if a particular unknown channel provides some edge cases that the pipeline may not capture, the pipeline could end up producing erroneous results.

---

I would agree with you in most cases but there is a `merge` channel that we do not use, so the script cannot halt due for this reason.